src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -56,6 +57,7 @@ class HMPPSAuthService(
       .retrieve()
       .onStatus({ HttpStatus.NOT_FOUND == it }, { Mono.just(null) })
       .bodyToFlux(AuthGroupResponse::class.java)
+      .retry(maxRetryAttempts)
```

Comment:
i wonder how we could get these 'auto-applied' to all calls for this client? i wonder if there's a way by adding a filter function to the base webclient ?

Reply:
Isn't this is a feature of the flux rather than the client consuming it.

Reply:
This is now changed to do retryWhen for TimeoutException and ConnectTimeoutException